Analysis.md

```diff
@@ -0,0 +1,78 @@
+
+
+
+## 2. Model
+
+![](https://lh6.googleusercontent.com/jn-FxfRtgjz9YBbW_PiRTHlkABWxkKVciUy6FafCa7qFvAwGUkQVM09QC9PcCTdJy-pSbd8DjqTIVHHkCQq85x9WcjPrB3bAkt3Y73e82oodQMA82IsT90DNKCnFS3ZpcVccaLXl)
+
+
+
+
+
+
+
+
+
+## 3. Class Responsibilities
+
+For each class, include an explanation about its responsibility and why you included it, mapping the classes to the system requirements.
+
+
+
+**User**
+
+* The User Class is responsible for keeping track of a given user's information with regards to how the backend will send them information.
+
+* It is included because it allows the necessary information pertaining to delivery information and account information to be stored in a single class, and managed easily from the backend.
+
+* It helps meet the system requirements because it stores how the user will receive their information.
+
+
+
+**Profile**
+
+* The Profile Class stores the preferences of a given user of DontPanic. This includes what kind of notifications they want to receive, and and how they want to receive them.
+
+* It is included to store the users personal preferences. While the User Class focuses on how our back end handles the delivery of information, the Profile Class focuses on what kind of information the user will receive, and how they want to receive it. It is the part that is customizable by them.
+
+* It meets the requirements by giving the user control of the information they want to receive and how they want to receive it.
+
+
+**Location**
+
+* Because our app uses geo-location, we need to add Location to our diagram.
+
+* This is because we send notifications that are area-specific, and are crucial to the use of our app. *Keeping these messages relevant to the user is a key part of our app and strongly aligns with our system requirements by keeping users informed with information related to their area.
+
+
+
+**Controller**
+
+* The Controller is what will functionally get the messages the user will receive. It does not store any data, but grabs the messages, which has the data needed.
+
+* Because the Message Class is data, we need something to get the message and deliver it to the user.
+
+* The Controller will help make the decisions. On which messages to recieve for the user, and will help keep messages relevant to preferences through keywords/tiers that are associated with Message Class.
+
+
+
+**Messages**
+
+* The Messages Class helps keep track of how messages are created, and the content in them.
+
+* Since each user can customize the kinds of messages they receive, a Message Class needed to be included to that a message can be tailored to their given preferences.

```

Comment:
```suggestion
* Since each user can customize the kinds of messages they receive, a Message Class needed to be included, to allow message classification, so that messages are only sent which meet given preferences.

```